src/core/thread/discover_scanner.cpp

```diff
@@ -336,12 +336,20 @@ void DiscoverScanner::HandleDiscoveryResponse(Mle::RxInfo &aRxInfo) const
     // Process MeshCoP TLVs
     while (offset < end)
     {
-        IgnoreError(aRxInfo.mMessage.Read(offset, meshcopTlv));
+        SuccessOrExit(error = aRxInfo.mMessage.Read(offset, meshcopTlv));
+
+        if (meshcopTlv.IsExtended())
+        {
+            SuccessOrExit(error = Tlv::ParseAndSkipTlv(aRxInfo.mMessage, offset));
+            continue;
```

Comment:
The other instances below have a `VerifyOrExit()` is that needed here too?

Reply:
Yes. Indeed. Thanks for noticing. Updated in new push.